Add internlm3 dense (#35694)
huggingface/transformers: docs/source/en/model_doc/internlm3.md

<!--Copyright 2024 The InternLM Team and The HuggingFace Team. All rights reserved.

Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
the License. You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
specific language governing permissions and limitations under the License.

⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
rendered properly in your Markdown viewer.

-->

# InternLM3

> ## Introduction

Reviewer: ```suggestion
## Overview
```

> InternLM3 has open-sourced an 8-billion parameter instruction model, InternLM3-8B-Instruct, designed for general-purpose usage and advanced reasoning. This model has the following characteristics:

Reviewer: ```suggestion
InternLM3 is an 8-billion parameter instruction model designed for general-purpose usage and advanced reasoning. This model has the following characteristics:
```

> - Enhanced performance at reduced cost: State-of-the-art performance on reasoning and knowledge-intensive tasks surpass models like Llama3.1-8B and Qwen2.5-7B. Remarkably, InternLM3 is trained on only 4 trillion high-quality tokens, saving more than 75% of the training cost compared to other LLMs of similar scale.

Reviewer: ```suggestion
- Enhanced performance at reduced cost: State-of-the-art performance on reasoning and knowledge-intensive tasks that surpass models like Llama3.1-8B and Qwen2.5-7B. Remarkably, InternLM3 is trained on only 4 trillion high-quality tokens, saving more than 75% of the training cost compared to other LLMs of similar scale.
```

> - Deep thinking capability: InternLM3 supports both the deep thinking mode for solving complicated reasoning tasks via the long chain-of-thought and the normal response mode for fluent user interactions.

Reviewer: ```suggestion
- Deep thinking capability: InternLM3 supports both deep thinking mode for solving complicated reasoning tasks with the long chain-of-thought technique and normal response mode for fluent user interactions.
```

## Usage tips

Model weights can be requested [here](https://huggingface.co/internlm/internlm3-8b-instruct).

> In the following, we demonstrate how to use `InternLM3-8B-Instruct` for the inference.

Reviewer: ```suggestion
In the following, we demonstrate how to use `InternLM3-8B-Instruct` for inference.
```

```python
>>> from transformers import AutoModelForCausalLM, AutoTokenizer
>>> device = "cuda" # the device to load the model onto

> >>> tokenizer = transformers.InternLM3Tokenizer.from_pretrained("internlm/internlm3-8b-instruct")
> >>> model = transformers.InternLM3ForCausalLM.from_pretrained("internlm/internlm3-8b-instruct", device_map="auto")

Reviewer: ```suggestion
>>> tokenizer = AutoTokenizer.from_pretrained("internlm/internlm3-8b-instruct")
>>> model = AutoModelForCausalLM.from_pretrained("internlm/internlm3-8b-instruct", device_map="auto")
```
